src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php

```diff
@@ -66,7 +73,11 @@ public function processRememberMe(RememberMeDetails $rememberMeDetails, UserInte
 
         [$series, $tokenValue] = explode(':', $rememberMeDetails->getValue());
         $persistentToken = $this->tokenProvider->loadTokenBySeries($series);
-        if (!hash_equals($persistentToken->getTokenValue(), $tokenValue)) {
+
+        if (
+            ($this->tokenVerifier && !$this->tokenVerifier->verifyToken($persistentToken, $tokenValue))
+            || !hash_equals($persistentToken->getTokenValue(), $tokenValue)
```

Comment:
Wouldn't it be better to swap the OR condition, given that `verifyToken` probably involves calling a db/cache and so adds more load on the server than the `hash_equals`?

Reply:
Oh sorry this was a mistake, I meant to only call the verifier if it is set, so that it is fully in control of token verification, and hash_equals is only there as fallback if no verifier is present. I dumbed the code down a bit now so it's clearer and does what I meant it to :)

Note that any sane implementation (and the two included) would call hash_equals first and if the token is valid (which it is in 99.9% of cases) it returns early so there is really no performance impact.